mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala

```diff
@@ -12,17 +12,23 @@ object SemVer {
       milestone: Option[Int]
   ) {
     def >(that: Version): Boolean = {
+      lazy val baseVersionEqual =
+        this.major == this.major && this.minor == that.minor && this.patch == that.patch
+
```

Comment:
Just a suggestion.

I've seen this part previously and was thinking to refactor it a bit if I will have to touch it.
Wdyt about replacing it on this impl?
```scala
  def <(that: Version): Boolean = {
      def toList(v: Version): List[Int] = {
        List(
          v.major,
          v.minor,
          v.patch,
          v.releaseCandidate.getOrElse(
            if (v.milestone.isEmpty) Int.MaxValue else 0
          ),
          v.milestone.getOrElse(
            if (v.releaseCandidate.isEmpty) Int.MaxValue else 0
          )
        )
      }

      val diff = toList(this)
        .zip(toList(that))
        .collectFirst {
          case (a, b) if a != b => a - b
        }
        .getOrElse(0)
      
      diff > 0
     }
```

Reply:
🤔 I honestly think the other one might be easier to read. It's definitely more verbose, but it's clear to me. I don't have strong feelings on it if we really want it changed, but I don't mind how it is now.

Reply:
np! feel free to ignore